Codecademy/gamut
# chore(AppHeader)!: Remove bookmarks and favorites

### Overview


Removes bookmarks and favorites code from the AppHeader.

_**This PR should not be merged until the day communicated to users that these features are being sunsetted.**_


### PR Checklist

- [x] Related to JIRA ticket: [WEB-2171]

#### Testing Instructions


Confirm all tests and checks pass here. See further testing instructions in https://github.com/codecademy-engineering/Codecademy/pull/33335 and https://github.com/codecademy-engineering/mono/pull/525

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/33335)  | [Monolith Env](https://pr-33335-monolith.dev-eks.codecademy.com) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/525)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-525)   |




[WEB-2171]: https://codecademy.atlassian.net/browse/WEB-2171?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
